### ISSUE TITLE
Wide gamut video get's incorrectly clipped to rec 709 on wide gamut monitor

### ISSUE THREAD

**Issue by vertexgamer** [mpv-player/mpv#9554]
- MPV build by shinchiro (20211128-git-f08db00)
- Win 10 with custom icc profile made with displaycal and wide gamut monitor

When playing a rec2020 hdr file, colors get clipped to rec 709 instead of fully using my monitor native gamut, so it looks like i'm viewing the content on an srgb monitor. Rec 709 content instead works perfectly and doesn't get over saturated.
I'm running these commands:       icc-profile-auto --audio-exclusive=yes

**Comment by haasn** [comment]
Can you attach the ICC profile?

**Comment by vertexgamer** [comment]
> Can you attach the ICC profile?
[MSI G271 #1 2021-08-16 01-49 0.3127x 0.329y 2.2 F-S XYZLUT+MTX.zip](https://github.com/mpv-player/mpv/files/7654309/MSI.G271.1.2021-08-16.01-49.0.3127x.0.329y.2.2.F-S.XYZLUT%2BMTX.zip)
sure! here it is

**Comment by vertexgamer** [comment]
@haasn 

i just tested mpv on another computer with an slightly wider than srgb monitor and a custom icc profile made with the shit spyderx software. Hdr/bt2020 colours looks good and no blacks are crushed. Could my issue be caused by the icc profile made with dysplaycal? Have you had any luck with the profile i sent you?

**Comment by haasn** [comment]
Sorry, I haven't looked at it yet. But the reason I asked for the ICC profile specifically is because I know it works on my end (with my ICC profile). So the only plausible conclusion is that the issue is somehow built into your ICC profile. Incidentally, mine is also generated by DisplayCAL. So either there's a regression in displaycal or you picked some weird settings that limit the input gamut?

**Comment by haasn** [comment]
Actually, can you attach the displaycal .log file?

**Comment by vertexgamer** [comment]
You mean the logs in C:\Users\Your Username\AppData\Roaming\DisplayCAL\logs ? I will send you all the logs made in 16/8. I made multiple profiles that day, the 01-49 is the one i'm currently using
[logs.zip](https://github.com/mpv-player/mpv/files/7656818/logs.zip)

**Comment by vertexgamer** [comment]
[DisplayCAL.log](https://github.com/mpv-player/mpv/files/7656828/DisplayCAL.log)
i almost forgot this

**Comment by aufkrawall** [comment]
I don't think you monitor is really wide gamut with just 86% DCI-P3. Your profile works in mpv with HDR content and does slightly desaturate:
![on](https://user-images.githubusercontent.com/13510490/144771174-25554dc1-7b52-479e-b10e-6202637cf0df.png)

![off](https://user-images.githubusercontent.com/13510490/144771177-0b818c56-8e1b-401b-ba92-88e587e1c0b6.png)

I've noticed there is also very little desaturation with Rec709 content with your profile, which might be odd for a 86% DCI-P3 device.

May I ask why you set black output offset to 0% in calibration settings? I suppose this prevents BT.1886 gamma boosting? I'd probably try that out then.

**Comment by vertexgamer** [comment]
> I don't think you monitor is really wide gamut with just 86% DCI-P3. Your profile works in mpv with HDR content and does slightly desaturate: ![on](https://user-images.githubusercontent.com/13510490/144771174-25554dc1-7b52-479e-b10e-6202637cf0df.png)
> 
> ![off](https://user-images.githubusercontent.com/13510490/144771177-0b818c56-8e1b-401b-ba92-88e587e1c0b6.png)
> 
> I've noticed there is also very little desaturation with Rec709 content with your profile, which might be odd for a 86% DCI-P3 device.
> 
> May I ask why you set black output offset to 0% in calibration settings? I suppose this prevents BT.1886 gamma boosting? I'd probably try that out then.
> 
> Edit: Ok, the "Video (D65, Rec. 1886)" preset does that.

- yes, my monitor doesn't fully cover dci p3, generally i focus on the reds, since it's my monitor strong point. On HDR i confirm what you have encountered, but i do find a noticeable difference on rec709 content.
- The reason i have choosen a 0% black offset, even if i like the intent of the feature, is due to the not so good black levels on my monitor, which caused the blacks to be clipped too much, making my photo editing more difficult and masking out/making less noticeable the noise in dark areas. I don't have access anymore to the hardware calibrator, so i can't make new profiles. 
- As you suggested i tryied using one of the less accurate profiles i made with the sh*t spiderx software, which uses the black offset by default, and apart from the usual gamut problem, the blacks appears to be working correctly, unfortunately now is the profile to be cutting them by default.
Sorry for long post

**Comment by ValZapod** [comment]
> 86% DCI-P3 device.

Is it dci-p3 or display-p3?? Because dci-p3 primaries are not something you can ever find on any consumer device, only in cinema Xenon projector. As for how to fix it, well 86% of display-p3 is more like BT.709. What is your EDID primaries, BTW? Did you compare to ICC profile before chromatic adaptation? It may be better than using D50 ICC hacky profiles.

**Comment by aufkrawall** [comment]
> Is it dci-p3 or display-p3?? Because dci-p3 primaries are not something you can ever find on any consumer device, only in cinema Xenon projector. 

You are technically correct, though DisplayCal refers to DCI-P3 when it comes to information about gamut coverage/volume.

I btw. did some tests and clamped gamut to rec709 scores just ~70% DCI-P3. Anyway, mpv seems to correct colors as intended here (also with a new Display-P3 ;) device I got in the meantime).
